packages/block-library/src/navigation/view.js

```diff
@@ -66,7 +66,14 @@ const { state, actions } = store(
 					actions.openMenu( 'hover' );
 			},
 			closeMenuOnHover() {
-				actions.closeMenu( 'hover' );
+				const { type, overlayOpenedBy } = getContext();
+				if (
+					type === 'submenu' &&
+					// Only open on hover if the overlay is closed.
```

Comment:
```suggestion
					// Only close on hover if the overlay is closed.
```

Reply:
Addressed in https://github.com/WordPress/gutenberg/pull/60162